csi/controller_server.go

```diff
@@ -780,39 +881,75 @@ func (cs *ControllerServer) DeleteSnapshot(ctx context.Context, req *csi.DeleteS
 	}
 
 	csiSnapshotType, sourceVolumeName, id := decodeSnapshotID(snapshotID)
-	if id == "" {
-		return nil, status.Errorf(codes.NotFound, "volume source snapshot %v is not found", snapshotID)
-	}
```

Comment:
Could you please explain why we should delete this check?

Reply:
the backing image has different snapshotID format
but we share the same decodeSnapshotID() function
- For BackingImage, the `sourceVolumeName`, `id` will be "", so we skip the check
- For Snapshot/Backup, we check in the below code https://github.com/longhorn/longhorn-manager/pull/1858/files#diff-4dbe40287acb148c5d0499c0adf977582db1e7b841f0d5d0943bd7e5e77b0cc8R890